evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -736,3 +737,44 @@ def test_component_graph_dataset_with_different_types():
     assert input_feature_names['Elastic Net'] == ['column_3', 'column_1_a', 'column_1_b', 'column_1_c', 'column_1_d',
                                                   'column_2_1', 'column_2_2', 'column_2_3', 'column_2_4', 'column_2_5', 'column_2_6']
     assert input_feature_names['Logistic Regression'] == ['Random Forest', 'Elastic Net']
+
+
+def test_component_graph_sampler():
+    graph = {'Imputer': [Imputer],
+             'OneHot': [OneHotEncoder, 'Imputer.x'],
+             'Undersampler': [Undersampler, 'OneHot.x'],
+             'Random Forest': [RandomForestClassifier, 'Undersampler.x', 'Undersampler.y'],
+             'Elastic Net': [ElasticNetClassifier, 'Undersampler.x', 'Undersampler.y'],
+             'Logistic Regression': [LogisticRegressionClassifier, 'Random Forest', 'Elastic Net']}
+
+    component_graph = ComponentGraph(graph)
+    component_graph.instantiate({})
+    assert component_graph.get_parents('Imputer') == []
+    assert component_graph.get_parents('OneHot') == ['Imputer.x']
+    assert component_graph.get_parents('Undersampler') == ['OneHot.x']
+    assert component_graph.get_parents('Random Forest') == ['Undersampler.x', 'Undersampler.y']
+    assert component_graph.get_parents('Elastic Net') == ['Undersampler.x', 'Undersampler.y']
```

Comment:
Interesting, I think I'm a little out of the loop with respect to the component graphs.  Is it expected that there's a branch from the undersampler to elastic net and rf? i assume because theyre both estimators that makes both estimators children of the last component?

That kind of leads me to another question : is this a test specific to the undersampler?  Or would the test look like this without the undersampler?

```
assert component_graph.get_parents('Imputer') == []
assert component_graph.get_parents('OneHot') == ['Imputer.x']
assert component_graph.get_parents('Random Forest') == ['OneHot.x', 'OneHot.y']
assert component_graph.get_parents('Elastic Net') == ['OneHot.x', 'OneHot.y']
```

If it would look like the above with OneHot taking the place of Undersampler, I think we really just need a more generalized test that tests the behavior with respect to the transformer/estimator boundary in the component graph without necessarily using the Undersampler.

Reply:
@chukarsten, currently, our component parents for all previous transformers only stores the `x` value in the `component_graph` parents' rather than storing both X and y since our previous transformers didn't alter `y`. I added in the support for `y` since our sampling methods (both over and under) will all be transforming both `X` and `y`, which is why the tests I added were specific to the undersampler. For your example, it would look like:

```python
assert component_graph.get_parents('Imputer') == []
assert component_graph.get_parents('OneHot') == ['Imputer.x']
assert component_graph.get_parents('Random Forest') == ['OneHot.x']
assert component_graph.get_parents('Elastic Net') == ['OneHot.x']
```

Reply:
So is this test a test that it's handling the passing forward of the y value?

Reply:
yep, this is making sure that when we add in an undersampler component to the pipeline, we keep track of both the X and y output of that component, while for other components we still only track the X output. This is definitely something that we can discuss in conjunction with your point about having the output of `transform()` be normalized among all transformers.